tracing-appender/src/non_blocking.rs

```diff
@@ -277,7 +277,16 @@ impl Drop for WorkerGuard {
                 // when the `Worker` calls `recv()` on a zero-capacity channel. Use `send_timeout`
                 // so that drop is not blocked indefinitely.
                 // TODO: Make timeout configurable.
-                let _ = self.shutdown.send_timeout((), Duration::from_millis(1000));
+                match self.shutdown.send_timeout((), Duration::from_millis(1000)) {
+                    Err(SendTimeoutError::Disconnected(_)) => (),
+                    Err(SendTimeoutError::Timeout(e)) => {
+                        println!("Failed to wait for logging worker shutdown. Error: {:?}", e);
+                    }
+                    Ok(_) => {
+                        // At this point it is safe to wait for `Worker` destruction without blocking
+                        self.handle.take().map(std::thread::JoinHandle::join);
```

Comment:
hmm, if the worker thread panicked (and `JoinHandle::join` returned an error), should we do something about that? we could at least print the error...

Reply:
13fec129c5666fe6cf0fe30053c4034a0ea72cc2